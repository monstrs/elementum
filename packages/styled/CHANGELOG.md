# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.1.7 (2019-02-22)


### Features

* **ts:** rewrite to TS ([6c92c9a](https://github.com/monstrs/elementum/commit/6c92c9a))
