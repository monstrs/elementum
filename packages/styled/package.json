{
  "name": "@elementum/styled",
  "version": "0.1.7",
  "license": "MIT",
  "main": "dist/index.js",
  "module": "dist/index.esm.js",
  "umd:main": "dist/index.umd.js",
  "typings": "dist/index.d.ts",
  "source": "src/index.ts",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "clean": "rimraf dist .rpt2_cache .rts2_cache_cjs .rts2_cache_es .rts2_cache_umd",
    "prepublishOnly": "yarn run build",
    "prebuild": "rimraf dist",
    "build": "microbundle --no-compress --sourcemap=false"
  },
  "dependencies": {
    "@elementum/modifiers": "0.1.7",
    "@emotion/core": "^10.0.7",
    "@emotion/styled": "^10.0.7"
  },
  "devDependencies": {
    "@types/node": "10.12.26",
    "@types/react": "16.8.4",
    "jest-emotion": "10.0.7",
    "microbundle": "0.9.0",
    "react": "16.8.3",
    "react-test-renderer": "16.8.3",
    "rimraf": "2.6.3"
  },
  "peerDependencies": {
    "react": "^16.8.2"
  }
}
