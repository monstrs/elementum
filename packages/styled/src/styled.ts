import { match, extract } from '@elementum/modifiers'
import emotionStyled from '@emotion/styled'
import { CreateStyledProxy } from './types'

const applyModifiers = styles => props =>
  Object.keys(styles).reduce((result, modifier) => {
    if (modifier === 'self') {
      return result
    }

    if (match(modifier, props)) {
      return {
        ...result,
        ...styles[modifier],
      }
    }

    return result
  }, {})

const styled: CreateStyledProxy = tag => styles => {
  const modifiers = extract(styles)

  return emotionStyled(tag, {
    shouldForwardProp: prop => !modifiers.includes(prop),
  })(styles.self, applyModifiers(styles))
}

export default styled
