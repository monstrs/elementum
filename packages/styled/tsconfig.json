{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "target": "es5",
    "module": "es2015",
    "allowJs": false,
    "strict": true,
    "sourceMap": true,
    "declaration": true,
    "declarationDir": "dist/types",
    "outDir": "dist/lib"
  },
  "include": ["src"],
  "extendedDiagnostics": true
}
