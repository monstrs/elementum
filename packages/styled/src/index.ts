import { CreateStyledProxyWithTags, CreateStyledProxy } from './types'
import { Tags } from './tags'
import styled from './styled'

const newStyled: CreateStyledProxy = styled.bind(null)

Object.keys(Tags).forEach(tagName => {
  newStyled[tagName] = newStyled(tagName as keyof JSX.IntrinsicElements)
})

export default newStyled as CreateStyledProxyWithTags
