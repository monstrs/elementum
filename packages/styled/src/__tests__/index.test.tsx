import React from 'react'
import serializer from 'jest-emotion'
import renderer from 'react-test-renderer'
import styled from '../index'

expect.addSnapshotSerializer(serializer)

describe('styled', () => {
  describe('snapshot base', () => {
    it('should match latest render snapshot', () => {
      const Base = styled.div({
        self: {
          background: 'red',
        },
      })

      const base = renderer.create(<Base />).toJSON()

      expect(base).toMatchSnapshot()
    })
  })
})
