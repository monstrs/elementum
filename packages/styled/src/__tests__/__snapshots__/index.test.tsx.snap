// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled snapshot base should match latest render snapshot 1`] = `
.emotion-0 {
  background: red;
}

<div
  className="emotion-0"
/>
`;
